packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx

```diff
@@ -322,39 +469,101 @@ const GroupForm = ({
     setRequirementSubForms([...allRequirements]);
   };
 
-  const validateSubForms = () => {
+  const validateSubForms = async () => {
     const updatedSubForms = [...requirementSubForms];
 
-    requirementSubForms.map((subForm, index) => {
-      try {
-        // HACK ALERT: this type of validation change should be done internally by zod, by we are doing this
-        // manually using javascript
-        const schema = getRequirementSubFormSchema(
-          subForm.values.requirementType,
-        );
-        if (subForm.values.requirementType === '') {
-          schema.pick({ requirementType: true }).parse(subForm.values);
-        } else {
-          schema.parse(subForm.values);
-        }
+    await Promise.all(
```

Comment:
would like @mzparacha to take a look at this

Reply:
Making `validateSubForms` async isn't a good idea in my opinion. This also gets used in the `onErrors` props of CWForm, meaning anytime CWForm gives an error `validateSubForms` gets called. This can happen as user types.

If our validation depends on any network/async/third-party dependency we should do it in the `onSubmit` explicitly and handle error as an API error (i.e then we can display it via toast for example)